modin/experimental/xgboost/test/test_xgboost.py

```diff
@@ -0,0 +1,175 @@
+)
+@pytest.mark.parametrize(
+    "num_actors",
+    [1, num_cpus, None, 400],
+    ids=["1", "num_cpus", "None", "400"],
+)
+@pytest.mark.parametrize(
+    "data",
+    [
+        (
+            load_iris(),
+            {
+                "max_depth": 3,
+                "eta": 0.3,
+                "silent": 1,
+                "objective": "multi:softprob",
+                "num_class": 3,
+            },
+        ),
+        (
+            load_diabetes(),
+            {
+                "colsample_bytree": uniform(0.7, 0.3),
+                "gamma": uniform(0, 0.5),
+                "learning_rate": uniform(0.03, 0.3),
+                "max_depth": randint(2, 6),
+                "n_estimators": randint(100, 150),
+                "subsample": uniform(0.6, 0.4),
+            },
+        ),
+        (load_digits(), {"max_depth": 6, "n_estimators": 200}),
+        (
+            load_wine(),
+            {
+                "objective": "multi:softprob",
+                "eta": 0.4,
+                "max_depth": 3,
+                "gamma": 1.0,
+                "num_class": 3,
+            },
+        ),
+        (
+            load_breast_cancer(),
+            {"objective": "binary:logistic", "eval_metric": ["logloss", "error"]},
+        ),
+    ],
+    ids=[
+        "load_iris",
+        "load_diabetes",
+        "load_digits",
+        "load_wine",
+        "load_breast_cancer",
+    ],
+)
+def test_dmatrix_train_predict_sklearn_datasets(data, num_actors, num_round):
+    dataset, param = data
+
+    X = dataset.data
+    y = dataset.target
+    dmatrix_xgb = xgboost.DMatrix(X, label=y)
+    with pytest.raises(AssertionError):
+        dmatrix_mxgb = xgb.DMatrix(X, label=y)
+
+    modin_X = pd.DataFrame(X)
+    modin_y = pd.Series(y)
+    dmatrix_mxgb = xgb.DMatrix(modin_X, label=modin_y)
+
+    evals_result_xgb = {}
+    evals_result_mxgb = {}
+    bst = xgboost.train(
+        param,
+        dmatrix_xgb,
+        num_round,
+        evals_result=evals_result_xgb,
+        evals=[(dmatrix_xgb, "train")],
+    )
+    modin_bst = xgb.train(
+        param,
+        dmatrix_mxgb,
+        num_round,
+        evals_result=evals_result_mxgb,
+        evals=[(dmatrix_mxgb, "train")],
+        num_actors=num_actors,
+    )
+
+    predictions = bst.predict(dmatrix_xgb)
+    modin_predictions = modin_bst.predict(dmatrix_mxgb, num_actors=num_actors)
+
+    if len(predictions.shape) > 1:
+        best_preds = np.empty((predictions.shape[0]))
+        modin_best_preds_ = np.empty((modin_predictions.shape[0]))
+        for i in range(predictions.shape[0]):
+            preds_val = 0
+            modin_preds_val = 0
+            for j in range(predictions.shape[1]):
+                preds_val += predictions[i][j]
+                modin_preds_val += modin_predictions[j][i]
+            best_preds[i] = preds_val
+            modin_best_preds_[i] = modin_preds_val
+        modin_best_preds = pd.DataFrame(modin_best_preds_)
```

Comment:
Why did you change previous approach in accuracy calculation? This approach gives `modin_best_preds` with column of `1` on the each dataset. This is not correct.

Reply:
this case was changed